packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,330 @@
-        has a url
-      </a>
+      <a href="#">has a url</a>
     </div>
   );
 
-  const { queryByTestId } = render(counterMock({ description: NotADescription, ...requiredAttributes }));
+  render(<CounterMock value={5} description={NotADescription} />);
 
-  expect(queryByTestId('test')).not.toBeInTheDocument();
+  expect(screen.queryByRole('link')).not.toBeInTheDocument();
 });
 
 test('accepts an Error Component', () => {
   const CustomError = (
     <FormControlError>
       This is a custom Error Component
-      <a href="#" data-testid="test">
-        has a url
-      </a>
+      <a href="#">has a url</a>
     </FormControlError>
   );
 
-  const { queryByTestId } = render(<FormGroup>{counterMock({ error: CustomError, ...requiredAttributes })}</FormGroup>);
+  render(
+    <FormGroup>
+      <CounterMock value={5} error={CustomError} />
+    </FormGroup>,
+  );
 
-  expect(queryByTestId('test')).toBeInTheDocument();
+  expect(screen.getByRole('link')).toBeInTheDocument();
 });
 
 test('does not accept non-Error Components', () => {
   const NotAnError = (
     <div>
       This is a not a custom error component
-      <a href="#" data-testid="test">
-        has a url
-      </a>
+      <a href="#">has a url</a>
     </div>
   );
 
-  const { queryByTestId } = render(<FormGroup>{counterMock({ error: NotAnError, ...requiredAttributes })}</FormGroup>);
+  render(
+    <FormGroup>
+      <CounterMock value={5} error={NotAnError} />
+    </FormGroup>,
+  );
 
-  expect(queryByTestId('test')).not.toBeInTheDocument();
+  expect(screen.queryByRole('link')).not.toBeInTheDocument();
 });
 
 test('renders both the add and subtract icons', () => {
-  const { getAllByRole } = render(counterMock(requiredAttributes));
+  render(<CounterMock value={5} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons.length).toBe(2);
+  expect(buttons).toHaveLength(2);
 });
 
 test('buttons are disabled when value hits max or min', () => {
-  const { getAllByRole, rerender } = render(counterMock({ ...requiredAttributes, value: 0 }));
+  const { rerender } = render(<CounterMock value={0} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons[0]).toHaveProperty('disabled', true);
-  expect(buttons[1]).toHaveProperty('disabled', false);
+  expect(buttons[0]).toBeDisabled;
+  expect(buttons[1]).toBeEnabled;
 
-  rerender(counterMock({ ...requiredAttributes, value: 10 }));
+  rerender(<CounterMock value={10} />);
 
-  expect(buttons[0]).toHaveProperty('disabled', false);
-  expect(buttons[1]).toHaveProperty('disabled', true);
+  expect(buttons[0]).toBeEnabled;
+  expect(buttons[1]).toBeDisabled;
 });
 
 test('value prop only accepts whole numbers', () => {
-  const { container } = render(counterMock({ max: 20, ...requiredAttributes }));
+  render(<CounterMock value={5} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
```

Comment:
No need to cast to `HTMLInputElement`:

<img width="387" alt="image" src="https://user-images.githubusercontent.com/4542735/149054371-a5da4c56-a144-428c-9479-f091f47f526b.png">